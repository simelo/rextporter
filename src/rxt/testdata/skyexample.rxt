
DATASET
    FOR SERVICE skycoin
    FOR STACK   skyfiber   

    DEFINE AUTH rest_csrf AS "skyauth"
        SET "url" TO "/api/v1/csrf"
        SET "method" TO "GET"
        SET "header" TO "X-CSRF-Token"
        SET "json_path" TO "csrf_token"

    GET forward_metrics FROM '/api/v2/metrics'
        SET "prefix" TO "skycoinexample"

    GET rest_api FROM '/api/v1/health'
        SET "auth" TO "skyauth"
        EXTRACT USING "jsonpath"
            METRIC
                NAME "skycoin_auth_csrf_enabled"
                TYPE GAUGE
                SET "path" TO "csrf_enabled"
            METRIC
                NAME "skycoin_blockchain_burn_factor"
                TYPE COUNTER
                SET "path" TO "user_verify_transaction.burn_factor"
            METRIC
                NAME "skycoin_blockchain_block_head"
                TYPE COUNTER
                SET "path" TO "blockchain.head.seq"

    GET rest_api FROM '/api/v1/network/connections'
        SET "auth" TO "skyauth"
        EXTRACT USING "jsonpath"
            METRIC
                NAME "skycoin_pex_connections"
                TYPE COUNTER
                LABELS "state", "outgoing", "listen_port", "user_agent",
                       "is_trusted", "burn_factor", "max_txn_size",
                       "max_decimals", "height"
                SET "path" TO "connections[*]"
                SET "label_path:is_trusted"   TO "is_trusted_peer"
                SET "label_path:burn_factor"  TO "unconfirmed_verify_transaction.burn_factor"
                SET "label_path:max_txn_size" TO "unconfirmed_verify_transaction.max_transaction_size"
                SET "label_path:max_decimals" TO "unconfirmed_verify_transaction.max_decimals"

