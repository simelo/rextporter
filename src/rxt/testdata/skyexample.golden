CTX LEX
EOL 0
KEY DATASET
BLK 4
KEY FOR SERVICE
VAR skycoin
EOL 4
KEY FOR STACK
VAR skyfiber
EOL 4
KEY DEFINE AUTH
VAR rest_csrf
KEY AS
STR "skyauth"
BLK 8
KEY SET
STR "url"
KEY TO
STR "/api/v1/csrf"
EOL 8
KEY SET
STR "method"
KEY TO
STR "GET"
EOL 8
KEY SET
STR "header"
KEY TO
STR "X-CSRF-Token"
EOL 8
KEY SET
STR "json_path"
KEY TO
STR "csrf_token"
EOB 8
EOL 4
KEY GET
VAR forward_metrics
KEY FROM
STR '/api/v2/metrics'
BLK 8
KEY SET
STR "prefix"
KEY TO
STR "skycoinexample"
EOB 8
EOL 4
KEY GET
VAR rest_api
KEY FROM
STR '/api/v1/health'
BLK 8
KEY SET
STR "auth"
KEY TO
STR "skyauth"
EOL 8
KEY EXTRACT USING
STR "jsonpath"
BLK 12
KEY METRIC
BLK 16
KEY NAME
STR "skycoin_auth_csrf_enabled"
EOL 16
KEY TYPE
KEY GAUGE
EOL 16
KEY SET
STR "path"
KEY TO
STR "csrf_enabled"
EOB 16
EOL 12
KEY METRIC
BLK 16
KEY NAME
STR "skycoin_blockchain_burn_factor"
EOL 16
KEY TYPE
KEY COUNTER
EOL 16
KEY SET
STR "path"
KEY TO
STR "user_verify_transaction.burn_factor"
EOB 16
EOL 12
KEY METRIC
BLK 16
KEY NAME
STR "skycoin_blockchain_block_head"
EOL 16
KEY TYPE
KEY COUNTER
EOL 16
KEY SET
STR "path"
KEY TO
STR "blockchain.head.seq"
EOB 16
EOB 12
EOB 8
EOL 4
KEY GET
VAR rest_api
KEY FROM
STR '/api/v1/network/connections'
BLK 8
KEY SET
STR "auth"
KEY TO
STR "skyauth"
EOL 8
KEY EXTRACT USING
STR "jsonpath"
BLK 12
KEY METRIC
BLK 16
KEY NAME
STR "skycoin_pex_connections"
EOL 16
KEY TYPE
KEY COUNTER
EOL 16
KEY LABELS
STR "state"
PNC ,
STR "outgoing"
PNC ,
STR "listen_port"
PNC ,
STR "user_agent"
PNC ,
STR "is_trusted"
PNC ,
STR "burn_factor"
PNC ,
STR "max_txn_size"
PNC ,
STR "max_decimals"
PNC ,
STR "height"
EOL 16
KEY SET
STR "path"
KEY TO
STR "connections[*]"
EOL 16
KEY SET
STR "label_path:is_trusted"
KEY TO
STR "is_trusted_peer"
EOL 16
KEY SET
STR "label_path:burn_factor"
KEY TO
STR "unconfirmed_verify_transaction.burn_factor"
EOL 16
KEY SET
STR "label_path:max_txn_size"
KEY TO
STR "unconfirmed_verify_transaction.max_transaction_size"
EOL 16
KEY SET
STR "label_path:max_decimals"
KEY TO
STR "unconfirmed_verify_transaction.max_decimals"
EOB 16
EOB 12
EOB 8
EOB 4
EOL 0
